src/ChainedValidator.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\XmlValidator;
+
+use DOMDocument;
+use function array_merge;
+
+final class ChainedValidator implements XmlValidator
```

Comment:
I saw [chain mostly used to handle a request only in one object](https://sourcemaking.com/design_patterns/chain_of_responsibility) but in this case all validators are always run to collect all the failures. Strictly speaking this is a [`Composite`](https://sourcemaking.com/design_patterns/composite)